phoenix-core/src/main/java/org/apache/phoenix/schema/transform/Transform.java

```diff
@@ -150,16 +156,52 @@ protected static void addTransform(
                         (changedProps.getColumnEncodedBytesProp() != null? changedProps.getColumnEncodedBytesProp() : table.getEncodingScheme()))
                 .build();
         SchemaExtractionProcessor schemaExtractionProcessor = new SchemaExtractionProcessor(systemTransformParams.getTenantId(),
-                connection.getQueryServices().getConfiguration(), newTable,  true);
+                connection.getQueryServices().getConfiguration(), newTable, true);
         String ddl = schemaExtractionProcessor.process();
         LOGGER.info("Creating transforming table via " + ddl);
         connection.createStatement().execute(ddl);
         upsertTransform(systemTransformParams, connection);
+
+        return newTable;
+    }
+
+    public static SystemTransformRecord getTransformRecord(Configuration conf, PTableType tableType, PName schemaName,
+                                                           PName tableName, PName dataTableName, PName tenantId,
+                                                           PName parentLogicalName) throws SQLException {
+        if (tableType == PTableType.TABLE) {
+            try (PhoenixConnection connection = QueryUtil.getConnectionOnServer(conf).unwrap(PhoenixConnection.class)) {
+                return Transform.getTransformRecord(schemaName, tableName, null, tenantId, connection);
+            }
+        } else if (tableType == INDEX) {
+            try (PhoenixConnection connection = QueryUtil.getConnectionOnServer(conf).unwrap(PhoenixConnection.class)) {
+                return Transform.getTransformRecord(schemaName, tableName, dataTableName, tenantId, connection);
+            }
+        } else if (tableType == VIEW) {
+            try (PhoenixConnection connection = QueryUtil.getConnectionOnServer(conf).unwrap(PhoenixConnection.class)) {
+                return Transform.getTransformRecord(SchemaUtil.getSchemaNameFromFullName(parentLogicalName.getString()),
+                        SchemaUtil.getTableNameFromFullName(parentLogicalName.getString()), null, tenantId==null? null:tenantId.getString(), connection);
+            }
+        }
+        return null;
+>>>>>>> 78571cc772... IndexRegionObserver changes for creating mutations for transforming table (#701)
```

Comment:
Looks like there's an incomplete merge or rebase here with conflicts left -- this probably doesn't compile

Reply:
Yeah, I fixed these and forgot to push. Sorry